sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainerProactiveInitConfigBuilder.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.models.CosmosContainerIdentity;
+
+import java.util.List;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+
+/**
+ * A builder to build {@link CosmosContainerProactiveInitConfig}
+ * */
+public final class CosmosContainerProactiveInitConfigBuilder {
+
+    private static final int MAX_NO_OF_PROACTIVE_CONNECTION_REGIONS = 5;
+    private final List<CosmosContainerIdentity> cosmosContainerIdentities;
+    private int numProactiveConnectionRegions;
+
+    /**
+     * Instantiates {@link CosmosContainerProactiveInitConfigBuilder}
+     *
+     * @param cosmosContainerIdentities the container identities - this parameter must be non-empty
+     */
+    public CosmosContainerProactiveInitConfigBuilder(List<CosmosContainerIdentity> cosmosContainerIdentities) {
+        checkArgument(
+            cosmosContainerIdentities != null && !cosmosContainerIdentities.isEmpty(),
+            "The list of container identities cannot be null or empty.");
+        this.cosmosContainerIdentities = cosmosContainerIdentities;
+    }
+
+    /**
+     * Sets the no. of regions to proactively connect to.
+     * <p>
+     * <br>The no of proactive regions to connect to belong to the preferred list of regions
+     * <br>In order to minimize latencies associated with warming up caches and opening connections
+     * the no. of proactive connection regions cannot be more
+     * than {@link CosmosContainerProactiveInitConfigBuilder#MAX_NO_OF_PROACTIVE_CONNECTION_REGIONS}.
+     * </p>
+     *
+     * @param numProactiveConnectionRegions the no of proactive connection regions
+     * @return Current {@link CosmosContainerProactiveInitConfigBuilder}
+     */
+    public CosmosContainerProactiveInitConfigBuilder setProactiveConnectionRegions(int numProactiveConnectionRegions) {
+        checkArgument(
```

Comment:
I think the check should be consistent (whether we allow 0 or not):

<img width="984" alt="image" src="https://user-images.githubusercontent.com/64233642/216421514-33dd6161-a0ef-4b15-8ee0-a8df6bfda891.png">

Reply:
Agreed - done